airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -1600,6 +1602,74 @@ public boolean getConnectionHasAlphaOrBetaConnector(final UUID connectionId) thr
     return countResult > 0;
   }
 
+  public Optional<ConnectorBuilderProject> getConnectorBuilderProject(final UUID builderProjectId, final UUID workspaceId) throws IOException {
+    return database.query(ctx -> ctx.select(CONNECTOR_BUILDER_PROJECT.asterisk())
+        .from(CONNECTOR_BUILDER_PROJECT)
+        .where(getBuilderProjectIdCondition(builderProjectId, workspaceId))
+        .fetch())
+        .map(DbConverter::buildConnectorBuilderProject)
+        .stream()
+        .findFirst();
+  }
+
+  public Stream<ConnectorBuilderProject> getConnectorBuilderProjectsByWorkspace(final UUID workspaceId) throws IOException {
+    final Condition matchByIds = CONNECTOR_BUILDER_PROJECT.WORKSPACE_ID.eq(workspaceId);
+
+    return database
+        .query(ctx -> ctx
+            .select(CONNECTOR_BUILDER_PROJECT.ID, CONNECTOR_BUILDER_PROJECT.WORKSPACE_ID, CONNECTOR_BUILDER_PROJECT.NAME,
+                CONNECTOR_BUILDER_PROJECT.ACTOR_DEFINITION_ID)
+            .from(CONNECTOR_BUILDER_PROJECT)
+            .where(matchByIds)
+            .fetch())
+        .map(DbConverter::buildConnectorBuilderProjectWithoutManifestDraft)
+        .stream();
+  }
+
+  public boolean deleteBuilderProject(final UUID builderProjectId, final UUID workspaceId) throws IOException {
+    return database.transaction(ctx -> ctx.deleteFrom(CONNECTOR_BUILDER_PROJECT))
+        .where(getBuilderProjectIdCondition(builderProjectId, workspaceId))
+        .execute() > 0;
+  }
+
+  public void writeBuilderProject(final ConnectorBuilderProject builderProject) throws IOException {
+    database.transaction(ctx -> {
+      final OffsetDateTime timestamp = OffsetDateTime.now();
+      final Condition matchId = CONNECTOR_BUILDER_PROJECT.ID.eq(builderProject.getBuilderProjectId());
```

Comment:
Should we use `getBuilderProjectIdCondition` here? If feels like there could be many `ConnectorBuilderProject` with the same project ID and different workspace based on how we delete [here](https://github.com/airbytehq/airbyte-platform/pull/44/files#diff-2d29b293778e290157be96e22294c2df1fdfa7c72a48961d03396776446deeddR1629)

Reply:
There should only be one row per project id as it's the PK - I changed how to delete in the follow-up PR here: https://github.com/airbytehq/airbyte-platform/pull/45